notebooks/Project 1 - Algorithms.ipynb

```diff
@@ -86,68 +86,74 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x108b80550>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
    "source": [
     "import time\n",
     "import math\n",
-    "import matplotlib\n",
     "import matplotlib.pyplot as plt\n",
-    "%matplotlib inline\n",
+    "import sys\n",
+    "sys.setrecursionlimit(2000)\n",
+    "% matplotlib inline\n",
     "\n",
     "# comparision of the time required for each alternative to run\n",
     "def iterative_factorial(n):\n",
     "    factorial = 1\n",
-    "    for i in range(n, 1, -1): # this sets range from the factorial (i.e. 5!) down to 1\n",
-    "        factorial *= n        # factorial is the variable the accumulates the product each loop\n",
+    "    for i in range(n, 1, -1):\n",
+    "        factorial *= n\n",
     "        n -= 1\n",
     "    return factorial\n",
     "\n",
+    "\n",
     "def recursive_factorial(n):\n",
-    "    if n<= 1:                 # exit out of the loop once we get down to 1\n",
+    "    if n < 1:\n",
     "        return 1\n",
-    "    return n * recursive_factorial(n-1) # core of the recursion\n",
+    "    return n * recursive_factorial(n - 1)\n",
     "\n",
-    "def factorial_compare(n):\n",
-    "    n_points.append(n) # build list of n for plot\n",
-    "    \n",
-    "    start_time_i = time.time() # start timestamp\n",
-    "    iterative_factorial(n)   # iterative case\n",
-    "    delta_t = time.time() - start_time_i  # elapsed time\n",
-    "    inter.append(delta_t)  # build list of results for plot\n",
-    "    \n",
-    "    start_time_r = time.time()\n",
-    "    recursive_factorial(n)  # recursive case\n",
-    "    delta_t = time.time() - start_time_r\n",
-    "    recur.append(delta_t)  # build list of results for plot\n",
-    "    \n",
-    "    start_time_p = time.time()\n",
-    "    math.factorial(n)      # added in the python built-in to show how optimzed it is, as results below show!\n",
-    "    delta_t = time.time() - start_time_p\n",
-    "    pyth.append(delta_t)  # build list of results for plot\n",
-    "    \n",
-    "# empty lists to collect data for plotting\n",
```

Comment:
Generally, it's better to contain like data in the same object, which is why I chose to store it in a dictionary (see below). This both reduces the number of lists you have to manage and reduces the number of write operations required.